tools/tsdb-chunks/main_test.go

```diff
@@ -0,0 +1,90 @@
+	tmpDir := t.TempDir()
+
+	spec := testutil.BlockSeriesSpec{
+		Labels: labels.FromStrings(labels.MetricName, "asdf"),
+		Chunks: []chunks.Meta{
+			tsdbutil.ChunkFromSamples([]tsdbutil.Sample{
+				sample{10, 11, nil, nil},
+				sample{20, 12, nil, nil},
+				sample{30, 13, nil, nil},
+			}),
+			tsdbutil.ChunkFromSamples([]tsdbutil.Sample{
+				sample{40, 0, tsdb.GenerateTestHistogram(1), nil},
+				sample{50, 0, tsdb.GenerateTestHistogram(2), nil},
+				sample{60, 0, tsdb.GenerateTestHistogram(3), nil},
+			}),
+		},
+	}
+
+	meta, err := testutil.GenerateBlockFromSpec(userID, tmpDir, []*testutil.BlockSeriesSpec{&spec})
+	require.NoError(t, err)
+
+	co := test.CaptureOutput(t)
+
+	chunkFilename := path.Join(tmpDir, meta.ULID.String(), "chunks", "000001")
+	require.NoError(t, printChunksFile(chunkFilename, true))
+
+	sout, _ := co.Done()
+
+	expected := `%s
+Chunk #0: position: 8 length: 15 encoding: XOR, crc32: 3d73a06f, samples: 3
+Chunk #0, sample #0: ts: 10 (1970-01-01T00:00:00.01Z), val: 11
+Chunk #0, sample #1: ts: 20 (1970-01-01T00:00:00.02Z), val: 12
+Chunk #0, sample #2: ts: 30 (1970-01-01T00:00:00.03Z), val: 13
+Chunk #0: minTS=10 (1970-01-01T00:00:00.01Z), maxTS=30 (1970-01-01T00:00:00.03Z)
+Chunk #1: position: 29 length: 56 encoding: histogram, crc32: 3e1d4429, samples: 3
+Chunk #1, sample #0: ts: 40 (1970-01-01T00:00:00.04Z), val: {count:18, sum:36.8, [-4,-2.82842712474619):2, [-2.82842712474619,-2):2, [-1.414213562373095,-1):3, [-1,-0.7071067811865475):2, [-0.001,0.001]:3, (0.7071067811865475,1]:2, (1,1.414213562373095]:3, (2,2.82842712474619]:2, (2.82842712474619,4]:2}
+Chunk #1, sample #1: ts: 50 (1970-01-01T00:00:00.05Z), val: {count:26, sum:55.199999999999996, [-4,-2.82842712474619):3, [-2.82842712474619,-2):3, [-1.414213562373095,-1):4, [-1,-0.7071067811865475):3, [-0.001,0.001]:4, (0.7071067811865475,1]:3, (1,1.414213562373095]:4, (2,2.82842712474619]:3, (2.82842712474619,4]:3}
+Chunk #1, sample #2: ts: 60 (1970-01-01T00:00:00.06Z), val: {count:34, sum:73.6, [-4,-2.82842712474619):4, [-2.82842712474619,-2):4, [-1.414213562373095,-1):5, [-1,-0.7071067811865475):4, [-0.001,0.001]:5, (0.7071067811865475,1]:4, (1,1.414213562373095]:5, (2,2.82842712474619]:4, (2.82842712474619,4]:4}
+Chunk #1: minTS=40 (1970-01-01T00:00:00.04Z), maxTS=60 (1970-01-01T00:00:00.06Z)
+`
+	expected = fmt.Sprintf(expected, chunkFilename)
+
+	require.Equal(t, expected, sout)
+}
+
+type sample struct {
+	t  int64
+	v  float64
+	h  *histogram.Histogram
+	fh *histogram.FloatHistogram
+}
+
+func (s sample) T() int64                      { return s.t }
+func (s sample) V() float64                    { return s.v }
+func (s sample) H() *histogram.Histogram       { return s.h }
+func (s sample) FH() *histogram.FloatHistogram { return s.fh }
+
+func (s sample) Type() chunkenc.ValueType {
+	switch {
+	case s.h != nil:
+		return chunkenc.ValHistogram
+	case s.fh != nil:
+		return chunkenc.ValFloatHistogram
+	default:
+		return chunkenc.ValFloat
+	}
+}
```

Comment:
This repeats in all new tests. Can we move it to some shared package for reuse (not sure if this is best place, but it may be a good start: `pkg/util/test`).

Same `sample` struct is also used in [compactor_test.go](https://github.com/grafana/mimir/blob/fbfc9f1b001d2a1bcf4379c37c7c9fc87a13d54f/pkg/compactor/compactor_test.go#L2048-L2072) or [gateway_test.go](https://github.com/grafana/mimir/blob/5ac2d2686b3a3e6b5c87c985cca5ace6970b2197/pkg/storegateway/gateway_test.go#L1599-L1631)

Reply:
Yep, I noticed the same! I think the best place for the shared code might actually be in the prometheus codebase though, so I opened https://github.com/prometheus/prometheus/pull/11948.

Reply:
If you want, we can merge this PR and then clean up `sample` type once after https://github.com/prometheus/prometheus/pull/11948 is available in Mimir.

Reply:
Sounds good to me!